frontend/src/metabase/static-viz/components/Gauge/GaugeContainer.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { Fragment } from "react";
+
+import { formatNumber } from "metabase/static-viz/lib/numbers";
+import { truncateText } from "metabase/static-viz/lib/text";
+import type { ColorGetter } from "metabase/static-viz/lib/colors";
+
+import {
+  START_ANGLE,
+  CHART_WIDTH,
+  GAUGE_OUTER_RADIUS,
+  CHART_VERTICAL_MARGIN,
+  GAUGE_INNER_RADIUS,
+  SEGMENT_LABEL_MARGIN,
+  DISTANCE_TO_MIDDLE_LABEL_ANCHOR,
+  MAX_SEGMENT_VALUE_WIDTH,
+  SEGMENT_LABEL_FONT_SIZE,
+} from "./constants";
+import {
+  removeDuplicateElements,
+  calculateRelativeValueAngle,
+  calculateSegmentLabelPosition,
+  calculateSegmentLabelTextAnchor,
+} from "./utils";
+
+import type { Card, Data, GaugeLabelData, Position } from "./types";
+import Gauge from "./Gauge";
+
+interface GaugeContainerProps {
+  card: Card;
+  data: Data;
+  getColor: ColorGetter;
+}
+
+export default function GaugeContainer({
+  card,
+  data,
+  getColor,
+}: GaugeContainerProps) {
+  const settings = card.visualization_settings;
+  const segments = settings["gauge.segments"];
+
+  const segmentMinValue = segments[0].min;
+  const segmentMaxValue = segments[segments.length - 1].max;
+
+  const center: Position = [
+    CHART_WIDTH / 2,
+    GAUGE_OUTER_RADIUS + CHART_VERTICAL_MARGIN,
+  ];
+
+  const value = data.rows[0][0];
+
+  const segmentMinMaxLabels: GaugeLabelData[] = segments
+    .flatMap(segmentDatum => {
+      return [segmentDatum.min, segmentDatum.max];
+    })
+    // gauge segments could be continuous i.e. the current max and the next min is the same value.
+    // So we should remove duplicate elements.
+    .reduce(removeDuplicateElements, [])
+    .map((segmentValue, index, segmentValues): GaugeLabelData => {
+      const isMinSegmentValue = index === 0;
+      const isMaxSegmentValue = index === segmentValues.length - 1;
+      const segmentValueAngle =
+        START_ANGLE +
+        calculateRelativeValueAngle(
+          segmentValue,
+          segmentMinValue,
+          segmentMaxValue,
+        );
+
+      if (isMinSegmentValue) {
```

Comment:
Both min and max values (e.g. 0 and 37,520) are rendered under the gauge.
![image](https://user-images.githubusercontent.com/1937582/194077373-0a47b483-47a2-4c71-852c-c336863c81a5.png)